src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -87,17 +89,17 @@ internal Task OnProjectChangedAsync(IProjectVersionedValue<IProjectSubscriptionU
             {
                 bool isDebuggable = await IsDebuggableAsync();
 
-                IVsStartupProjectsListService startupProjectsListService = await _startupProjectsListService.GetValueAsync();
+                IVsStartupProjectsListService? startupProjectsListService = await _startupProjectsListService.GetValueAsync();
```

Comment:
If this can return `null`, then it should be annotated correctly, which would have caught this.

One option is to define `_startupProjectsListService` as having type `IVsService<IVsStartupProjectsListService?>`.

The other option is to annotate `IVsService<T>.GetValueAsync(...)` as returning `Task<T?>`. @davkean I think we discussed this before and the thinking was that most services are expected to be present. However the current approach means we have to remember to add a nullable type parameter here. There are only 14 usages of that method in our solution. Now that we've had more time with C# nullable reference types, do you feel differently? I suggest we change the return to nullable and introduce `Assumes.Present` calls for services we expect to be always present. This will create a safer default behaviour for new usages of `IVsService<>`.

Reply:
I would like to figure out why this returns null, instead of hiding it and introducing a bug where we no longer show up in the Startup list. This would affect every single project in the solution.

This service should always be present, we need to figure out why its not.

Reply:
I still feel strongly that this should be non-nullable in most cases. If we checked for null here, we would be hiding a bug and telemetry wouldn't be telling us about it.